src/process_generator.c

```diff
@@ -25,21 +33,48 @@ int main(int argc, char *argv[])
     get_scheduling_algo(schedulerConfig);
 
     // 3. Initiate and create the scheduler and clock processes.
+
+    start_program("scheduler.out");
+    start_program("clk.out");
+
     // 4. Use this function after creating the clock process to initialize clock
-    // initClk();
+    initClk();
     // To get time use this
     // int x = getClk();
     // printf("current time is %d\n", x);
     // TODO Generation Main Loop
     // 5. Create a data structure for processes and provide it with its parameters.
     // 6. Send the information to the scheduler at the appropriate time.
+    
+    msgq_id = msgget(SHKEY, 0666 | IPC_CREAT);
+    msgbuf_t msgbuf;
+
+    while (!is_queue_empty(processes_queue))
+    {
+        process_info_t *process_data = (process_info_t *)front(processes_queue);
+        while (process_data && process_data->arrival == getClk())
+        {
+            msgbuf.mytype = getClk();
+            msgbuf.message = (*process_data);
+
+            msgsnd(msgq_id, &msgbuf, sizeof(msgbuf.message), IPC_NOWAIT);
+            
+            dequeue(processes_queue);
+            process_data = (process_info_t *)front(processes_queue);
+        }
+    }
+
```

Comment:
when the generator sends a collections of processes at this time step, it should send a SIGUSR1 signal to the scheduler to initiate the handling function
Note: it must send the signal only after sending all the processes that arrived at this time step